src/ripple/protocol/jss.h

```diff
@@ -562,6 +562,7 @@ JSS(sub_index);             // in: LedgerEntry
 JSS(subcommand);            // in: PathFind
 JSS(success);               // rpc
 JSS(supported);             // out: AmendmentTableImpl
+JSS(sync);                  // in: Submit
```

Comment:
If I am correct that this becomes part of our client-facing API, we should make sure this is the name we want.  I find myself wondering if another name, such as `submit` or maybe `submit_mode` would be better.  I'm not the right person to answer this question.  I'm simply trying to make this decision more visible.  If `sync` is the best name, I'm fine with that.

Reply:
@intelliot what's the best way to define this new API field? I like it as is, since the behavior has to do with being synchronous. But I also don't care enough either way. How should we handle this?

Reply:
1. Can you (re)explain the purpose and meaning of the field?
2. Why does it need to be user-facing? 
3. How should users decide what value to set in the field?

Once we have simplified and accurate answers to the above, then we should collect feedback from API users like @justinr1234, @mvadari, @mDuo13 to get a recommendation for how to define and introduce the field.

Reply:
@mDuo13 @mayurbhandary can you please review the proposed API change, described at the top of this PR?

Reply:
@mvadari ^

Reply:
Confirmed with Mark:

1. The purpose of the proposed `sync` field in the `submit` API is to allow users to control whether the server processes transactions immediately, or defers their processing to a later time.
2. It's user-facing because we are keeping the behavior the same for existing users. New implementations should use "async" or "wait" for overall improved throughput. "sync" can still be used since it is not a problem at current tx volumes; there does not seem to be any need to rush a change.
3. Users should choose:
  a. "wait" if they want to know about certain transaction submission errors that occur when a transaction is being applied. Compared to "sync", transaction application may be slightly delayed (up to 100ms), but this makes it easier for the server to keep up with the network.
  b. If they want to maximize throughput, then they should use "async". The server responds as quickly as possible, but since the transaction hasn't been applied yet, the client won't know about certain errors during transaction processing, if they occur.

"sync" may be a confusing name for the field since it is also one of its potential values ("sync"). Perhaps @mDuo13, @mvadari, or someone else could suggest a better name (if they are convinced that it makes sense to add the field at all, of course).

Reply:
I like `submit_mode` or maybe  just `mode` (since it would be a param to a "submit" command, it would be a "submit mode" either way).

Reply:
@ximinez what about "sync_mode"?

Reply:
Sync mode sounds confusing to me. I prefer mode or submit_mode.

Reply:
I could see the value in `sync_mode`. Like I said earlier, we already know that it's related to submit, but what happens if down the road there's some _other_ processing option that needs a separate mode. I have no idea what that would be, but you never know. If we do `sync_mode` now, then it'll be less confusing to add `foo_mode` later.

Reply:
"sync_mode" means that the particular mode defines the behavior having to do with synchronicity.
"mode" can be anything and "submit_mode" is redundant. On the other hand, the interface is something that strikes everybody differently. What do you guys think at this point? @justinr1234 @ximinez

Reply:
`sync_mode` works for me.